tools/serve/serve.py

```diff
@@ -778,20 +780,23 @@ def run(**kwargs):
 
     setup_logger(config["log_level"])
 
+    ssl_environment = get_ssl_environment(config)
```

Comment:
Shouldn't this still be used as a context manager?

Reply:
I know there's *some* difference because removing the `with` didn't work for one of the other cases last week, but it works here. Does it matter, what kinds of things should one use `with` for?

Reply:
With is used in RAII situations, usually where you have some setup you want to do on entering the block and some resource that you want to be automatically freed at the end of the block.

In this case I think some ssl types generate temporary files we create on `__enter__` and delete on `__exit__`. So those will break. It's a pity it's not more obvious that something can break if you use this without running `__enter__`. Presumably various patterns could support that but we aren't using them here.

Reply:
So I guess I should just put back `with` then.

Reply:
Done.